src/System.CommandLine/StringExtensions.cs

```diff
@@ -173,13 +174,81 @@ bool CanBeUnbundled(string arg)
 
                 return arg.StartsWith("-") &&
                        !arg.StartsWith("--") &&
-                       arg.RemovePrefix()
-                          .All(CharacterIsValidOptionAlias);
+                       TryUnbundle(arg.RemovePrefix(), out replacement);
+
+                Token TokenForOptionAlias(char c) =>
+                    argumentDelimiters.Contains(c) 
+                    ? null
+                    : knownTokens.SingleOrDefault(t => 
+                        t.Type == TokenType.Option && 
+                        t.Value.RemovePrefix() == c.ToString());
```

Comment:
I don't believe this check is valid since it is possible for tokens to differ by only their prefix. This does raise the question of what should be done when this case occurs.

Reply:
This is how the old functionality worked though? Alternative prefixes should maybe disable unbundeling?

Reply:
@jonsequitur any feedback on what is best here? This looks like an area where POSIX behavior conflicts with the desire to make invocation east for command line app user.

Reply:
I think it's reasonable to only support unbundling on `"-"` and not on other prefixes.

Reply:
So this can be considered resolved?

Reply:
I think so, yes.